packages/data/src/resolvers-cache-middleware.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * creates a middleware handling resolvers cache invalidation.
+ *
+ * @param {Object} registry
+ * @param {string} reducerKey
+ *
+ * @return {function} middleware
+ */
+const createResolversCacheMiddleware = ( registry, reducerKey ) => () => ( next ) => ( action ) => {
+	const resolvers = registry.select( 'core/data' ).getCachedResolvers( reducerKey );
+	Object.entries( resolvers ).forEach( ( [ selectorName, resolversByArgs ] ) => {
+		const resolver = get( registry.namespaces, [ reducerKey, 'resolvers', selectorName ] );
+		if ( ! resolver || ! resolver.shouldInvalidate ) {
+			return;
+		}
+		resolversByArgs.forEach( ( value, args ) => {
+			if ( value !== false || ! resolver.shouldInvalidate( action, ...args ) ) {
```

Comment:
I'm having trouble following what actually is the shape of `resolversByArgs`. When is `value` `false` and why does that have an impact on invalidation? It's not obvious in the unit tests.

Reply:
`resolversByArgs` is the map storing the cache resolution status for a given selector.

`Map([ args ] => boolean)`

If the value is `false` it means this resolver has finished its resolution which means we need to invalidate it, if it's `true` it means it's inflight and the invalidation is not necessary.

Reply:
That reads like a good inline comment.